streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBGenericOptionsToDbOptionsColumnFamilyOptionsAdapter.java

```diff
@@ -610,15 +617,21 @@ public AccessHint accessHintOnCompactionStart() {
         return dbOptions.accessHintOnCompactionStart();
     }
 
-    @Override
+    @Deprecated
     public Options setNewTableReaderForCompactionInputs(final boolean newTableReaderForCompactionInputs) {
-        dbOptions.setNewTableReaderForCompactionInputs(newTableReaderForCompactionInputs);
+        String message = "This method has been removed from the underlying RocksDB. " +
+                "It was not affecting compaction even in earlier versions. " +
+                "It is currently a no-op method.";
+        log.warn(message);
         return this;
     }
 
-    @Override
+    @Deprecated
     public boolean newTableReaderForCompactionInputs() {
-        return dbOptions.newTableReaderForCompactionInputs();
+        String message = "This method has been removed from the underlying RocksDB. " +
+                "It is now a method which always return false.";
```

Comment:
```suggestion
                "It is now a method which always returns false.";
```

Reply:
Ah, apologies, I will get into the habit of running all of these before I upload new commits.